frontend/src/app/office-hours/office-hours.models.ts

```diff
@@ -0,0 +1,214 @@
+}
+
+export interface TicketDetails {
+  id: number;
+  oh_event: OfficeHoursEvent;
+  type: TicketType;
+  state: TicketState;
+  description: string;
+  have_concerns: boolean;
+  caller_notes: string;
+  created_at: Date;
+  called_at: Date;
+  closed_at: Date;
+  creators: SectionMember[];
+  caller: SectionMember | null;
+}
+
+export interface TicketDraft {
+  oh_event: OfficeHoursEventPartial;
+  description: string;
+  type: TicketType;
+  creators: { id: number }[]; // check this type ?
+}
+
+export interface TicketPartial {
+  id: number;
+  oh_event: OfficeHoursEventPartial | null;
+  type: TicketType | null;
+  state: TicketState | null;
+  description: string | null;
+  have_concerns: boolean | null;
+  caller_notes: string | null;
+  created_at: Date | null;
+  called_at: Date | null;
+  closed_at: Date | null;
+  creators: SectionMember[] | null;
+  caller: SectionMember | null;
+}
+
+export interface OfficeHoursEvent {
+  id: number;
+  oh_section: OfficeHoursSection;
+  room: Room;
+  type: OfficeHoursEventType;
+  mode: OfficeHoursEventModeType;
+  description: string;
+  location_description: string;
+  event_date: string;
+  start_time: string;
+  end_time: string;
+}
+
+export interface OfficeHoursEventPartial {
+  id: number;
+  oh_section: OfficeHoursSection | null;
+  room: Room | null;
+  type: OfficeHoursEventType | null;
+  mode: OfficeHoursEventModeType | null;
+  description: string | null;
+  location_description: string | null;
+  event_date: string | null;
+  start_time: string | null;
+  end_time: string | null;
+}
+
+export interface OfficeHoursEventDraft {
+  oh_section: OfficeHoursSectionPartial;
+  room: RoomPartial; // should this be a room or room partial?
+  type: OfficeHoursEventType;
+  mode: OfficeHoursEventModeType;
+  description: string;
+  location_description: string;
+  event_date: string;
+  start_time: string;
+  end_time: string;
+}
+
+export enum Weekday {
+  Monday = 0,
+  Tuesday = 1,
+  Wednesday = 2,
+  Thursday = 3,
+  Friday = 4,
+  Saturday = 5,
+  Sunday = 6
+}
+
+// Method to get the abbreviated name of the weekday
+export namespace Weekday {
+  export function getAbbreviatedName(day: Weekday): string {
+    switch (day) {
+      case Weekday.Monday:
+        return 'Mon';
+      case Weekday.Tuesday:
+        return 'Tue';
+      case Weekday.Wednesday:
+        return 'Wed';
+      case Weekday.Thursday:
+        return 'Thu';
+      case Weekday.Friday:
+        return 'Fri';
+      case Weekday.Saturday:
+        return 'Sat';
+      case Weekday.Sunday:
+        return 'Sun';
+      default:
+        return '';
+    }
+  }
+}
+
+export interface RoomPartial {
+  id: string;
+}
+
+export interface OfficeHoursEventDetails {
+  id: number;
+  oh_section: OfficeHoursSection;
+  room: Room;
+  type: OfficeHoursEventType;
+  mode: OfficeHoursEventModeType;
+  description: string;
+  location_description: string;
+  event_date: string;
+  start_time: string;
```

Comment:
To make conversions easier, we have separated `..Json` models from regular models, where the `..Json` models store dates and strings, and then the regular models store these fields as dates. There is a fromJson function which converts it. This might help to prevent having to convert dates everywhere else in the code.